### PR TITLE
dev-libs/mini-xml: bump to version 2.12

### PR DESCRIPTION
**Pull request by ConiKost**
Also bumped to EAPI=7.

Closes: https://bugs.gentoo.org/668604
Signed-off-by: Conrad Kostecki <conrad@kostecki.com>
Package-Manager: Portage-2.3.50, Repoman-2.3.11

### CODE REVIEW

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+EAPI="7"
+
+inherit autotools
+
+MY_P="${P/mini-xml/mxml}"
+
+DESCRIPTION="A small XML parsing library that you can use to read XML data files or strings"
+HOMEPAGE="https://github.com/michaelrsweet/mxml
+	https://www.msweet.org/mxml/"
+SRC_URI="https://github.com/michaelrsweet/mxml/releases/download/v${PV}/${MY_P}.tar.gz"
```
Comment (Zlogene) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: there is standard https://github.com/michaelrsweet/mxml/archive/v2.12.tar.gz

so I do not know why rhe whole ranaming game is needed. but `-> ${P}.tar.gz`

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+LICENSE="Mini-XML"
+SLOT="0"
+IUSE="static-libs test threads"
```
Comment (Zlogene) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: you do not use the `test` use flag

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+DEPEND="virtual/pkgconfig"
```
Comment (Zlogene) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: if this is EAPI=7 this is BDEPEND

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 1 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Hm, good question. Thanks, changed now.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 2 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Whoops, removed.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 3 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Yeah thanks! Fixed.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 4]
```diff
@@ -0,0 +1,57 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI="7"
+
+inherit autotools
+
+MY_PN="mxml"
```
Comment (mgorny) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: I'm a bit confused. If upstream's using 'mxml' everywhere, why are we naming it 'mini-xml'?

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+MY_P="${MY_PN}-${PV}"
+
+DESCRIPTION="A small XML parsing library that you can use to read XML data files or strings"
+HOMEPAGE="https://github.com/michaelrsweet/mxml
+	https://www.msweet.org/mxml/"
+SRC_URI="https://github.com/michaelrsweet/${MY_PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~ppc ~ppc64 ~sparc ~x86"
+LICENSE="Mini-XML"
```
Comment (mgorny) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: The license in the git repo has a newer date. Could you diff them?

**Comment by mgorny** [comment]
Also:
1. Static library is built in `src_install` instead of `src_compile`,
2. Please make the Makefile rules verbose.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 5 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Yes. Thanks for the hint. There are two differences.

1) It's now under LGPL2 instead of LGPL.
2) Linking with it must use LGPL2 also.

In this case, I will update the license directly, since it already was part of 2.10 (oldest version in portage).

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 4 above)
Comment (mgorny) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: @gentoo/qa, your opinion? Should we continue calling it 'mini-xml' or rename to 'mxml'?

Comment (Zlogene) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: @mgorny other distros are using mxml, do not know why we are different

**Comment by ConiKost** [comment]
> Static library is built in src_install instead of src_compile,
> Please make the Makefile rules verbose.

Both fixed.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 4 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: I've added an commit, for renaming from `dev-libs/mini-xml` to `dev-libs/mxml` and all reverse deps.

Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Can someone explain me the failed CI?

**Comment by mgorny** [comment]
You didn't rename ebuild files.

**Comment by ConiKost** [comment]
Thx!

File: licenses/Mini-XML [hunk 6]
```diff
@@ -1,26 +1,26 @@
 			   Mini-XML License
```
Comment (mgorny) [inline licenses/Mini-XML]: Could you also mention that the new license applies to all current versions in the commit message? This will help others understand why the change was done in place.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+
+	# Don't run always tests
+	# Enable verbose compiling
+	sed -e '/ALLTARGETS/s/testmxml//g' -e '/.SILENT:/d' -i Makefile.in || die
+
+	# Build only static-libs, when requested by user, also build docs without static-libs in that case
+	if ! use static-libs; then
+		local mysedopts=(
+			-e '/^install:/s/install-libmxml.a//g'
+			-e '/^mxml.xml:/s/-static//g'
```
Comment (mgorny) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: Does this mean that with USE=static-libs, installed executables are linked statically? We really do want to use dynamic linking *always*.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+			-e '/^mxml.epub:/s/-static//g'
+			-e '/^valgrind/s/-static//g'
+			-e '/.\/mxmldoc-static/s/-static//g'
+		)
+		sed ${mysedopts[@]} -i Makefile.in || die
```
Comment (mgorny) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: Missing quoting around mysedopts.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+	fi
+
+	default
```
Comment (mgorny) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: You want to `default` before the `sed`s. This will ensures that patches made against vanilla upstream sources will apply cleanly.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+	eautoconf
+}
+
+src_configure() {
+	local myeconfopts=(
+		$(use_enable threads)
+		--with-docdir=/usr/share/doc/${P}
```
Comment (mgorny) [inline dev-libs/mini-xml/mini-xml-2.12.ebuild]: PF

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 7 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: No, all executables are still dynamic. Let me try to explain.

The Makefile ist just pretty shitty. It always builds static and dynamic lib.
When I disable compilation of static lib, aka USE=-static-libs, I have to patch arround:
1) Tests are always run and do compile with static libs for itself, so it's build, even, when I don't want that. So I patched this.
2) The creation of docs uses an selfbuild binary (it's never installed to the system), to create the docs. Per default, it's static. So again, it forces building the static lib, even, If a user don't want to. So patching this also around.

File: licenses/Mini-XML (hunk 6 above)
Comment (ConiKost) [reply licenses/Mini-XML]: Done.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 8 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Done.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 9 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Done.

File: dev-libs/mini-xml/mini-xml-2.12.ebuild (hunk 10 above)
Comment (ConiKost) [reply dev-libs/mini-xml/mini-xml-2.12.ebuild]: Done.

File: licenses/Mini-XML (hunk 6 above)
Comment (mgorny) [reply licenses/Mini-XML]: Portage is a package manager, and it doesn't contain Mini-XML.

Comment (ConiKost) [reply licenses/Mini-XML]: Ok, changed.

**Comment by ConiKost** [comment]
Hm, I can't reproduce. When I do completly uninstall `dev-libs/mini-xml` and run a fresh install, it just works. It does not matter, if with `FEATURES="test"` and/or `USE="static-libs"`. Any ideas?

**Comment by mgorny** [comment]
Maybe preserved-libs keeps the lib for you?
